### ISSUE TITLE
Peer Review Draft analysis (Gruppe C)

### ISSUE THREAD

**Issue by NadineBraun** [Abschlussprojekt-Weatherdata/project#23]
**Peer review team**

- Peer review by: Gruppe C
- Names of team members that participated in this review: Nadine Braun, Alena Vetter

**Goal of the project**

- _Lineare Regression:_ Ziel ist es, die Temperatur anhand den Windgeschwindigkeit 
vorherzusagen. Grundlage hierfür sind historische Wetterdaten. Außerdem soll die generelle 
Beziehung zwischen Temperatur und Wind (Windstärke, Windrichtung und 
Windgeschwindigkeit) näher analysiert werden. Darüber hinaus wird untersucht, ob die 
Vorhersage der Temperatur mithilfe von Windgeschwindigkeit akkurater ist, als mithilfe
multipler Faktoren, wie z.B. Luftdruck, Schneefall etc.

- _Klassifikation:_ Kann mithilfe der logistischen Regression das Auftreten extremer und normaler 
Wetterverhältnisse mithilfe der Temperatur bzw. der Windgeschwindigkeit vorhergesagt 
werden. Außerdem soll geprüft werden, ob mithilfe eines binären Klassifikationsmodells 
Extremwetter, wie z.B. Stürme, vorhergesagt werden kann. Darüber hinaus soll untersucht 
werden, ob Extremwetter generell mithilfe multipler Faktoren vorhergesagt werden kann. 
Grundsätzlich soll also in der linearen Modellierung die Temperatur mithilfe verschiedener 
Aspekte vorhergesagt werden, während die Klassifikation den Fokus auf die Vorhersage von 
Extremwetter legt. In jedem liegen historische Wetterdaten dem Modell zugrunde.

**Data**

- Das verwendete Dataset enthält Wetterdaten, die vom European Centre for 
Medium Range Weather Forecasts (ECMWF)zur Verfügung gestellt wurden. 

- Es besteht aus meteorologischen Daten der Jahre 2015-2022 aus der Region Bancroft in Ontario (Kanada). 

- Das Dataset besteht aus verschiedenen Wettermerkmalen, wie Temparatur, Wind, Luft und Niederschlag, 
wobei die Wetterdaten stündlich aufgezeichnet und niedergeschrieben wurden. 

- Jeder Datensatz repräsentiert somit die Wetterverhältnisse einer vollen Stunde. 

- Insgesamt hat das Dataset einen Umfang von 65345 Observations und 184 Spalten. 

- Für die Vorhersage von Wetterphänomenen, entsprechend dem Ziel dieses Projekts, ist der Datensatz geeignet, 
da er umfassende meteorologische Daten über eine längere Zeitreihe enthält.

**Approach, tools and methods**

Zunächst wird mithilfe von deskriptiver Statistik die Struktur und die Verteilung der Daten dargestellt,
um ein besseres Verständnis für den Inhalt des Datasets sowie für die Beziehung der verschiedenen
untereinander zu schaffen. Hierfür wurden beispielsweise Scatterplotts, Histogramme und Boxplot
erstellt.

- _Regressionsanalyse | Temperatur und Wind:_ Für die Vorhersage der Temperatur mithilfe von
Windgeschwindigkeit wurden vier verschiede Modelle herangezogen und deren Effektivität
verglichen: Lineare Regression, Gradient Boosting Regressor, Stochastic Gradient Descent
Regressor und Support Vector Regressor. Anschließend werden die Daten in Trainings- und
Testdaten aufgeteilt, die Modelle gefitted und mit den Trainigsdaten trainiert, bevor die Güte
der vier Modelle mithilfe von Mean Squared Error (MSE) und Mean Absolut Error (MAE)
verglichen wird.

- _Multiple lineare Regression | Temperatur und atmosphärische Einflussgrößen:_ Zunächst 
werden die Zusammenhänge aller Variablen durch eine Korrelationsmatrix dargestellt.
Anschließend wird ein lineares Regressionsmodell gefittet und die Relevanz der einzelnen
Variablen mithilfe von Forward Selection und Backwards Feature Elimination analysiert. Als
Vergleichsgröße wird der R^2 Score herangezogen. Anschließend wird ein weiteres lineares
Regressionsmodell erstellt, diesmal lediglich mit den Predictoren Windgeschwindigkeit und
Windrichtung. Die Güte des Modells wird anhand MSE, MAE und R^2 bestimmt. Auf gleiche
Weise wird die Windgeschwindigkeit anhand multipler Predictorvariablen modelliert und
bewertet. Für weitergehende Analysen, wie beispielsweise saisonale Schwankungen, werden
darüber hinaus SARIMAX und XGBoost verwendet, diese werden mit verschiedenen
Kennzahlen, wie AIC bewertet.

- _Logistische Regression | Extremwetter und Temperatur:_ Auch hier wird das Dataset zunächst 
in Trainigs- und Testdaten geteilt. Anschließend wird das logistische Regressionsmodell
angewendet und an den Testdaten evaluiert. Hierfür werden Kennzahlen wie AIC, Precision,
Recall und F1-Score herangezogen. Mithilfe der Confusion Matrix wird zunächst ein Überblick
über die Modellergebnisse gegeben.

- _Binäre Klassifikation | Extremwetter:_ Das Wetter wird in zwei Werte unterteilt: Blue Sky und
Extremwetter. Als nächstes werden die Kennzahlen zahlreicher Modell tabellarisch verglichen.
Die besten Modell werden im Anschluss weiterverwendet: ExtraTreesClassifier, XGBClassifier,
RandomForestClassifier, and LGBMClassifier. Die Qualität der einzelnen Modelle wird dann
anhand der oben genannten Kennzahlen bewertet.

- _Multiclass Klassifikation | Spezifisches Extremwetter:_ Hier wir das oben betrachtete
Extremwetter in insgesamt 7 Extremwettertypen unterteilt. Es wurden insgesamt vier
Modelle angewendet: KNN, SVM, DTC, GBC. Anhand der Confusion Matrix und
Classification Report wurden die Ergebnisse der verschiedenen Modell dargestellt und
bewertet.

**Lack of clarity**

- Gerade bei vielen Predictorvariablen ist es an einigen Stellen nicht ganz offensichtlich, welche Variablen genau 
im Modell berücksichtigt werden. (z.B. Binäre Extremwetter)

- Beim Modell zur Vorhersage des Windes, wird im Modell selbst die Windgeschwindigkeit genutzt, in der Überschrift 
und Beschreibung hingegen wird Bezug auf die Windrichtung genommen. Hier ist unklar, was genau die 
Responsevariable sein soll.

- Außerdem ist uns der Zusammenhang der Fragen, die mithilfe des Modells beantwortet werden 
sollen, nicht ganz klar, da viele von ihnen in eine ähnliche Richtung gehen sich teilweise nur durch die 
Anzahl der Predictorvariablen unterscheiden. Sind es damit wirklich verschiedene Fragestellungen 
oder eher verschiedene Ebenen des gleichen Modells?

**Possible improvements**

- Bei der Klassifizierung zwischen Extremwetter und Blue Sky erscheint die Grenze sehr drastisch. Durch 
die Unterteilung entsteht ein großes Ungleichgewicht in der Verteilung der Datensatzanzahl, was sich 
negativ auf die Modellgüte auswirken kann.

- Im Projekt wurden viele verschiedene Modelle angewendet und verglichen. Eventuell könnte es aber
auch sinnvoll sein, zur Verbesserung der Modellgüte Anpassungen vorzunehmen, wie z.B. Eliminierung 
von Ausreißern oder Reduzierung der Predictorvariablen.

- Aufgrund der Masse an verwendeten Variablen werden viele Darstellung sehr klein oder überladen. 
Es könnte an einigen Stellen Sinn machen, nochmal näher auf reduzierte Darstellung zurückzugreifen.

- Beispielsweise bei dem Abschnitt mit den Histogrammen stimmt die Reihenfolge der Beschreibungen im Text 
nicht mit der Reihenfolge der Diagramme überein, was den Überblick und die Zuordnung 
erschwert, auch wenn die Beschriftung eindeutig ist.

- Beschriftung und Lesbarkeit Diagramme: Unter 4.2.3 sind beispielsweise die Beschriftung der Piecharts nicht lesbar. 
Hier wäre möglicherweise eine Legende sinnvoll. Bei 5.4.2.2 könnte ein anderer Winkel bei der Beschriftung 
der x-Achse zur Lesbarkeit beitragen.

- Im Data Setup sollten nach unserem Verständnis alle notwendigen Bibliotheken und Module 
importiert werden. Dies erfolgt derzeit immer an der Stelle, an der der Import benötigt wird.

- Die Aufzählung unter der Überschrift bei 5.1 ist durchgängig mit 1. beschriftet.

- Die Formatierung der Überschriften ist z.T. nicht durchgängig.

- Unter dem Punkt Data Structure könnte noch df.info() und df.tail() ergänzt werden.

- Unter dem Code zu Data Corrections könnte noch eine Auskommentierung der vorgenommenen 
Korrekturen ergänzt werden.

**Presentation**

- Was ist euer Gesamtfazit zu den Ergebnissen eures Projekts und den Grenzen, auf die eure Modelle 
gestoßen sind?

- Was würdet ihr anders machen, wenn ihn nochmal anfangen würdet? Würdet ihr euch auf die 
gleichen Fragestellungen fokussieren?

- Gab es Zusammenhänge in den Wetterdaten, die euch besonders überrascht haben

**Organization**

- Durch die Verwendung zahlreicher und verschiedener Modelle im Zusammenhang mit dem großen 
Umfang des Projekts fällt es zunehmend schwer, dem roten Faden der Arbeit zu folgen. Hierfür wäre
eventuell ein Inhaltsverzeichnis für einen schnellen Überblick sinnvoll.

- Die verwendeten Modelle können im Ordner models nicht gefunden werden. Deshalb sind wir nicht 
ganz sicher, ob die Modelle vielleicht nicht gespeichert wurden, es könnte aber auch ein 
Berechtigungsthema sein.

- Die Codekommentierungen helfen gut bei der Nachvollziehbarkeit der Codes. Allerdings sind die 
Codezeilen zum Teil sehr lang, hier könnte man den Code eventuell auf mehrere Schritte aufteilen 
(vgl. z.B. 5.3.4.5 Temperature Prediction (Temporal): Visualize results)

**Further comments**
Die Hinweise zur verwendeten Literatur mithilfe von Fußnoten am Ende eines Abschnitts anzugeben,
trägt sehr zur Übersichtlichkeit des Textes bei und ermöglicht einen ungestörten Lesefluss.
